pumpkin-solver/src/containers/key_value_heap.rs

```diff
@@ -43,11 +43,29 @@ impl<Key: StorageKey, Value> Default for KeyValueHeap<Key, Value> {
     }
 }
 
+impl<Key, Value> KeyValueHeap<Key, Value> {
+    pub(crate) const fn new() -> Self {
+        Self {
+            values: Vec::new(),
+            map_key_to_position: KeyedVec::new(),
+            map_position_to_key: Vec::new(),
+            end_position: 0,
+        }
+    }
+}
+
 impl<Key, Value> KeyValueHeap<Key, Value>
 where
     Key: StorageKey + Copy,
     Value: AddAssign<Value> + DivAssign<Value> + PartialOrd + Default + Copy,
 {
+    /// Get the keys in the heap.
+    ///
+    /// The order in which the keys are yielded is unspecified.
+    pub(crate) fn keys(&self) -> impl Iterator<Item = Key> + '_ {
+        self.map_position_to_key.iter().copied()
```

Comment:
This now also includes keys which are not in the heap; should probably slice until `end_position`